### PR TITLE
docs: Update templates and guides

### PR DESCRIPTION
**Pull request by castastrophe**
### What has changed and why

Summarize files edited as part of this MR along with a brief description of what was changed/why.

* Added more Issue and PR templates
* Added code of conduct
* Added contribution guidelines

### Ready-for-merge Checklist

Check off items as they are completed.  Feel free to delete items if they are not applicable to your PR.

- [x] Expected files: all files in this pull request are related to one request or issue (no stragglers or scope-creep).
- [x] Documentation updates are clear and concise.
- [x] Changelog updated.

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/new_component.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+#### Browser requirements
+
+Your component should work in all of the following environments:
+
+- [ ] Latest 2 versions of Edge
+- [ ] Internet Explorer 11 (should be useable, not pixel perfect)
+- [ ] Latest 2 versions of Firefox (one on Mac OS, one of Windows OS)
+- [ ] Firefox 60.7.2 (or latest version for Red Hat Enterprise Linux distribution)
```
Comment (starryeyez024) [inline .github/PULL_REQUEST_TEMPLATE/new_component.md:46]: the latest for RHEL is 68, should we modify this?

File: .github/PULL_REQUEST_TEMPLATE/new_component.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+- [ ] Latest 2 versions of Chrome (one on Mac OS, one of Windows OS)
+- [ ] Latest 2 versions of Safari
+- [ ] Galaxy S9 Firefox
+- [ ] iPhone X Safari
```
Comment (starryeyez024) [inline .github/PULL_REQUEST_TEMPLATE/new_component.md:50]: could we simply these mobile checkboxes to "Android mobile device" / "Apple mobile device"

File: .github/PULL_REQUEST_TEMPLATE/new_component.md (hunk 1 above)
Comment (castastrophe) [inline .github/PULL_REQUEST_TEMPLATE/new_component.md:46]:
```suggestion
- [ ] Firefox 68 (or latest version for Red Hat Enterprise Linux distribution)
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+Please note that this project is released with a [Contributor Covenant Code of Conduct](CODE_OF_CONDUCT.md). By participating in this project you agree to abide by its terms.
+
+## Submitting a pull request
+
+1. [Fork][fork] and clone the repository (see the Quick start in the [README][README.md])
+2. Create a new branch.  Please use the [conventional commit](cc) formatting for the branch name.
+    > `git checkout -b feat-my-branch-name`
+3. Make your change, [add tests](https://patternfly.github.io/patternfly-elements/develop/step-3/), and make sure the automated testing passes.
+4. Push to your fork and [submit a pull request][pr].
```
Comment (starryeyez024) [inline CONTRIBUTING.md:21]: should this list mention use of a PR template?

File: .github/PULL_REQUEST_TEMPLATE/new_component.md (hunk 2 above)
Comment (castastrophe) [reply .github/PULL_REQUEST_TEMPLATE/new_component.md:50]: Sure!

File: .github/PULL_REQUEST_TEMPLATE/new_component.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+Your component should work in all of the following environments:
+
+- [ ] Latest 2 versions of Edge
+- [ ] Internet Explorer 11 (should be useable, not pixel perfect)
+- [ ] Latest 2 versions of Firefox (one on Mac OS, one of Windows OS)
+- [ ] Firefox 68 (or latest version for Red Hat Enterprise Linux distribution)
+- [ ] Latest 2 versions of Chrome (one on Mac OS, one of Windows OS)
+- [ ] Latest 2 versions of Safari
+- [ ] Galaxy S9 Firefox
```
Comment (castastrophe) [inline .github/PULL_REQUEST_TEMPLATE/new_component.md:49]:
```suggestion
- [ ] Android mobile device (such as the Galaxy S9)
```

File: .github/PULL_REQUEST_TEMPLATE/new_component.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+- [ ] iPhone X Safari
```
Comment (castastrophe) [inline .github/PULL_REQUEST_TEMPLATE/new_component.md:50]:
```suggestion
- [ ] Apple mobile device (such as the iPhone X)
```

File: CONTRIBUTING.md (hunk 3 above)
Comment (castastrophe) [reply CONTRIBUTING.md:21]: Sure, do you have any ideas about how to word that?

Comment (castastrophe) [reply CONTRIBUTING.md]: I took a stab at this

**Pull request merged**